irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -253,14 +292,25 @@ namespace iroha {
           log_(logger::log("PostgresQueryExecutor")) {}
 
     QueryExecutorResult PostgresQueryExecutor::validateAndExecute(
-        const shared_model::interface::Query &query) {
+        const shared_model::interface::Query &query,
+        const bool validate_signatories = true) {
```

Comment:
ping^
fix tests instead, that should be enough

Reply:
Ping again, here's more detailed description
That interface is quite ugly and case for `validate_signatories == false` is used only in tests. Thus it's make sense to make the following function signature: `validateAndExecute(const shared_model::interface::Query &query)` and delegate unsafe executing to a new (virtual & probably protected) function, that doesn't perform any validation. In tests the last one should be called via inheritance or whatever

Reply:
This is rather an issue with tests, since signatories should always be validated as a part of query. Current interface will be more easily refactored when tests will be fixed.